### ISSUE TITLE
Support for custom sidecars

### ISSUE THREAD

**Issue by Philmod** [mattermost/mattermost-server#16876]
#### Summary
In order to use Cloud SQL for Mattermost, in a secure way, they recommend adding a proxy as a sidecar. 
https://cloud.google.com/sql/docs/mysql/connect-kubernetes-engine#running_the_proxy_as_a_sidecar

Unfortunately, it's not possible today as the Operator rewrites the whole deployment as soon as it's modified.

**Comment by balan2010** [comment]
Hi! I would like to work on this

**Comment by Szymongib** [comment]
Hi @balan2010, sure feel free to take it!
If you need any support let us know here or on [Mattermost community server](https://community.mattermost.com/).

**Comment by balan2010** [comment]
Thanks @Szymongib  sure will get in touch in community channel if I need any support :-)

**Comment by balan2010** [comment]
@Szymongib  would you be able to assign this to me. I don't have permission to do so

**Comment by vineelyalamarthy** [comment]
@jwilander  @Philmod  @balan2010  is there help needed on MatterMost operator.

**Comment by Philmod** [comment]
@balan2010 Any update on this? Thanks for picking it up!

**Comment by vineelyalamarthy** [comment]
Is any one already working on this? I would like to contribute. @Philmod  @balan2010

**Comment by balan2010** [comment]
@vineelyalamarthy I won't be able to work on this for atleast couple of months. So feel free to work on it

**Comment by jasonblais** [comment]
@vineelyalamarthy I have assigned the issue to you

**Comment by vineelyalamarthy** [comment]
@jasonblais  is there a grooming or vetting process that happens before a posted issue becomes a story that we work on. 
Also do we want to support  Cloud SQL for Mattermost as asked by @Philmod.

**Comment by jasonblais** [comment]
Yes, we have a process to review tickets before turning them to help wanted: https://handbook.mattermost.com/contributors/contributors/ways-to-contribute/help-wanted

@Szymongib I would defer to you regarding the Cloud SQL question?

**Comment by Szymongib** [comment]
I think we would like to have a more generic mechanism that allows adding sidecars to pods running Mattermost and Cloud SQL would be just one option.

**Comment by Philmod** [comment]
Any update here?

**Comment by Philmod** [comment]
Hi! Any status here?

**Comment by gabrieljackson** [comment]
Hey @Philmod, I was just shown this request and think we can add support for custom sidecars fairly easily. I will give you an update in the next few days.

**Comment by gabrieljackson** [comment]
@Philmod the functionality to add custom sidecars to the Mattermost deployment was merged. Note that this is the original implementation of sidecars and not the [new feature-gated one](https://kubernetes.io/blog/2023/08/25/native-sidecar-containers) which is in alpha, although we may migrate to that in the future once it stabilizes more.

This will be included in our next operator release which will be made available after we complete testing.

**Comment by Philmod** [comment]
Thanks @gabrieljackson! Is there any documentation on how to configure such a sidecar, maybe from an `installation.mattermost.com` kubernetes config?

Right now the [doc](https://docs.mattermost.com/install/install-kubernetes.html) explains how to set up the SQL connection secret for mattermost to connect, but would you know how we could adapt that if we wanted to use the [Cloud SQL Auth Proxy](https://cloud.google.com/sql/docs/mysql/connect-kubernetes-engine#proxy)?
